src/helpers/assistant/tools.ts

```diff
@@ -0,0 +1,231 @@
+    intervalAmount: z.number().describe("The number of periods to aggregate over. Together with intervalType this can be 1 Days or 3 Minutes."),
+    intervalType: z.enum(["Minutes", "Hours", "Days", "Weeks", "Months"])
+      .describe("The type of interval to aggregate over. Together with intervalAmount this can be 1 Days or 3 Minutes."),
+    aggregateFunctions: z.array(z.enum(["sum", "avg", "count", "min", "max"])).describe("The aggregations functions to apply to the granular data.")
+  });
+
+  name = "queryDeviceDataV2Aggregate";
+
+  description = `Can query a participant's aggregated device data. This can be used to get for instance a participant's hourly steps 
+    count or their min max heart rate every day.`;
+
+  async _call(input: z.infer<typeof this.schema>) {
+    let response = await MyDataHelps.queryDeviceDataV2Aggregate(input as DeviceDataV2AggregateQuery);
+
+    return JSON.stringify(response.intervals.map(({ date, statistics }) => ({ date, statistics })));
+  }
+}
+
+export class QueryDailySleepTool extends StructuredTool {
+  schema = z.object({
+    namespace: z.enum(["Fitbit", "AppleHealth"])
+      .describe("The namespace of the device data, representing the manufacturer of the devices used to collect the data."),
+    type: z.string()
+      .describe("The device data type is specific to the namespace. For Apple Health this is called Sleep Analysis, for Fitbit it is called Sleep."),
+    observedAfter: z.string().optional()
+      .describe("The start of the date range for the query. This is a date (no time) in the participant's local time. This is inclusive."),
+    observedBefore: z.string().optional()
+      .describe("The end of the date range for the query. This is a date (no time) in the participant's local time. This is inclusive.")
+  });
+
+  name = "queryDeviceDataV2DailySleep";
+
+  description = `Can query daily aggregated sleep data. This will include sleep broken down by sleep stage.
+    Use this function when you need to do sleep aggregate queries instead of deviceDataV2Aggregate.`;
+
+  async _call(input: z.infer<typeof this.schema>) {
+    let response = await MyDataHelps.queryDeviceDataV2DailySleep(input as DeviceDataV2AggregateQuery);
+
+    return JSON.stringify(response.sleepStageSummaries.map(({ date, duration, source, value }) => ({ date, duration, source, value })));
+  }
+}
+
+export class QueryDailyDataTool extends StructuredTool {
+  schema = z.object({
+    type: z.string().describe("The type of daily data to query."),
+    startDate: z.string().describe("The start of the date range for the query. This is a datetime in the participant's local timezone."),
+    endDate: z.string().describe("The end of the date range for the query. This is a datetime in the participant's local timezone.")
+  });
+
+  name = "queryDailyData";
+
+  description = "Query daily data for a participant.";
```

Comment:
I suspect we may want to hint this a little better; I'll have to play with it a bit more to see what its like when I don't have the V2 data types enabled, but in many cases I suspect we want the assistant to be using this rather than the V2 endpoints...because the v2 endpoints are only available in certain projects and are more "raw".

How about getting rid of the "GetAllDailyDataTypesTool" and instead embedding all the daily data types in the hints for the tool so it knows to look here for that data?  (I might not fully be understanding how it's working though or how it decides what tools to use)

Reply:
Thoughts on this?  Again like this is going to be the best source of daily data that will match graphs in most cases, so I think we should give it all the data types this can supply

Reply:
For `queryDeviceDataV2` I included in the description:
```
Before using this tool call the getDeviceDataV2AllDataTypes tool to see which data types are available. If a particular data type is not available use the queryDailyData tool instead.
```